OrcScala/src/orc/API.scala

```diff
@@ -57,80 +64,318 @@ trait OrcRuntimeProvides {
   */
 trait OrcRuntimeRequires extends InvocationBehavior
 
-/** Define invocation behaviors for a runtime
+/** This Invoker or Accessor should be inlined into the caller if supported.
```

Comment:
I see your point. And this interface could be moved to PorcE. I added it since there may be a HUGE win for site to be able to tell Truffle that they can me inlined. But honestly that might be a premature optimization. I'll remove this for now and assume they cannot be inlined aggressively. We'll revisit this later if I need that information.